app/js/lib/intent-parser/confirmation.js

```diff
@@ -0,0 +1,170 @@
+/* global TwitterCldr, TwitterCldrDataBundle */
+
+'use strict';
+
+import moment from 'components/moment';
+import 'components/cldr/en';
+import 'components/cldr/core';
+
+/*
+ * @todo:
+ *   * @see http://www.unicode.org/cldr/charts/29/verify/dates/en.html
+ *     for formatting the time of the day.
+ */
+
+const p = Object.freeze({
+  // Properties
+  listFormatter: Symbol('listFormatter'),
+
+  // Methods
+  formatUser: Symbol('formatUser'),
+  formatAction: Symbol('formatAction'),
+  formatTime: Symbol('formatTime'),
+  isToday: Symbol('isToday'),
+  isTomorrow: Symbol('isTomorrow'),
+  isThisMonth: Symbol('isThisMonth'),
+  formatHoursAndMinutes: Symbol('formatHoursAndMinutes'),
+});
+
+const DEFAULT_LOCALE = 'en';
+const PATTERNS = {
+  en: {
+    pattern: `OK, I'll remind [users] [action] [time].`,
+    formatUser: (user) => user
+      .replace(/\bme\b/gi, 'you')
+      .replace(/\bI\b/gi, 'you')
+      .replace(/\bmy\b/gi, 'your')
+      .replace(/\bmine\b/gi, 'yours'),
+  },
+  fr: {
+    pattern: `OK, je rappelerai [users] [action] [time].`,
+    formatUser: (user) => user,
+  },
+  ja: {
+    pattern: `承知しました。[time][users]に[action]をリマインドします。`,
+    formatUser: (user) => user,
+  },
+};
+
+export default class Confirmation {
+  constructor(locale = DEFAULT_LOCALE) {
+    this.locale = locale;
+
+    TwitterCldr.set_data(TwitterCldrDataBundle);
+
+    this[p.listFormatter] = new TwitterCldr.ListFormatter();
+  }
+
+  /**
+   * Generate a phrase to be spoken to confirm a reminder.
+   *
+   * @param {Object} reminder
+   * @return {string}
+   */
+  getReminderMessage(reminder) {
+    console.log(reminder);
+
+    let locale = this.locale;
+    if (!PATTERNS[this.locale] || !PATTERNS[this.locale].pattern) {
```

Comment:
We could extract this out into a method, or property getter to avoid duplicating it.